plugins/woocommerce/src/Internal/Admin/Marketing.php

```diff
@@ -180,124 +165,15 @@ public function component_settings( $settings ) {
 			return $settings;
 		}
 
-		$settings['marketing']['installedExtensions'] = InstalledExtensions::get_data();
+		/**
+		 * InstalledExtensions helper class.
+		 *
+		 * @var InstalledExtensions $installed_extensions
+		 */
+		$installed_extensions = wc_get_container()->get( InstalledExtensions::class );
 
-		return $settings;
-	}
+		$settings['marketing']['installedExtensions'] = $installed_extensions->get_data();
```

Comment:
I'd like to suggest a defensive check here to make sure that `$installed_extensions` is an instance of `InstalledExtensions`.

```suggestion
		if ( $installed_extensions instanceof InstalledExtensions ) {
			$settings['marketing']['installedExtensions'] = $installed_extensions->get_data();
		}
```

Reply:
Hmm. I don't think that's necessary here because it's not user input, and we can safely assume that the container will always return the correct object. I also don't see any instances in the code that perform this check when they obtain an object from the container.

Reply:
> we can safely assume that the container will always return the correct object

I thought that wasn't always a given. But if we can safely assume the container returns the object we want, then the additional check is not needed